src/host.rs

```diff
@@ -54,6 +125,72 @@ impl Host {
         }
     }
 
+    pub(crate) fn accept(&mut self) -> Option<Envelope> {
+        let now = Instant::now();
+        let deque = &mut self.listener.as_mut()?.deque;
+
+        match deque.front() {
+            Some(Envelope {
+                instructions: DeliveryInstructions::DeliverAt(time),
```

Comment:
There are three hosts, A, B & C. A can't reach B, but tries to connect. C also tries to connect to B. What happens?

I think this is broken, because syn(A:B) is at the front of the deque but is not eligible for delivery. If so, please open an issue. When we fix it, we should add a regression test.

Reply:
That sounds broken to me, but I'll write the test and report back to verify. It shouldn't be too tough to fix if it is.